### ISSUE TITLE
freerdp not trusting certificate with target host in the certificate SAN

### ISSUE THREAD

**Issue by rvigliotti-sf** [FreeRDP/FreeRDP#7720]
I get a certificate warning every time I connect to my Remote Desktop Service Environment, even if the certificate is trusted. I believe the reason is that the hostname I'm connecting to is in the SAN of the cert and not in the Subject.

I do not get any warning using the MSFT Mac RDP client or the Windows one.

This happens with both xfreerdp and RoyalTSX

**Comment by akallabeth** [comment]
We present a warning whenever one of the following is not met:
1. Certificate trusted by system root ca
2. Certificate in freerdp `known_hosts` file for the server connecting to
3. Certificate Subject (or any of the alternate names) matches server connecting to 

So, if you really connect to the correct hostname (not just IP) and the certificate is not self signed and trusted by your system ca, please add some logs so we can check.

**Comment by rvigliotti-sf** [comment]
@akallabeth I forgot to mention that this only happens when I try to connect to a RDS Environment with Gateway and Broker. If I directly RDP to a single host in the same environment (hence, same PKI infra) I don't get any warning.

I'll try to provide some useful anonymous logs

**Comment by akallabeth** [comment]
@rvigliotti-sf ah, ok, so you get the certificate warning for the redirected target machine?

**Comment by rvigliotti-sf** [comment]
@akallabeth Correct, and from latest tests seems only when using RoyalTSX and not freerdp

**Comment by akallabeth** [comment]
@rvigliotti-sf ok, that explains it.

That is `expected` behavior for `FreeRDP` as we have no real way of checking the destination certificate. (you usually connect via `IP` or `netbios name` and so you get mismatches on the certificate name or you even have self signed certificates on the destination machines)

(We do not implicitly trust everything beyond the gateway)

also note: If your specific issue is `RoyalTSX` only (check you don´t already have accepted the certificate with `xfreerdp` and is in the `known_hosts` file) better also contact them.
